operator/pkg/helmreconciler/prune.go

```diff
@@ -116,8 +116,8 @@ func (h *HelmReconciler) PruneControlPlaneByRevisionWithController(iopSpec *v1al
 	}
 	pids, err := proxy.GetIDsFromProxyInfo("", "", iopSpec.Revision, ns)
 	if err != nil {
-		return errStatus,
-			fmt.Errorf("failed to check proxy infos: %v", err)
+		// maybe istiod not run, we should ingnore the err to avoid delete bolcking
+		scope.Errorf("failed to check proxy infos: %v", err)
```

Comment:
I don't think it's secure to ignore the error here, at least we should check if the error is no istiod instance running.
We have that type of error, see: https://github.com/istio/istio/blob/0947addd23e3e90dc327ed84494c80c9d1006972/pkg/kube/client.go#L791

Reply:
I would agree with @morvencao , check the errors if it is indeed, no istiod exists.

Reply:
ok  i  will check it ,if this is the error,then ignore it